src/statsig.erl

```diff
@@ -1,140 +1,48 @@
 -module(statsig).
 
--behaviour(gen_server).
+-behaviour(application).
 
 -export(
   [
-    init/1,
-    terminate/2,
-    handle_call/3,
-    handle_cast/2,
-    handle_info/2,
-    check_gate/3,
+    start_link/0,
+    start/2,
+    stop/1,
+    check_gate/2,
+    log_event/3,
     log_event/4,
-    log_event/5,
-    flush/1
+    flush/0
   ]
 ).
 
--import(utils, [get_timestamp/0]).
--import(network, [start/0, stop/0, request/4]).
--import(evaluator, [eval_gate/3]).
--import(logging, [get_event/4, get_exposure/4]).
+start_link() ->
+  gen_server:start_link(?MODULE, [], []).
 
-init([{apiKey, ApiKey}]) ->
-  network:start(),
-  Body = network:request(ApiKey, "download_config_specs", #{}, true),
-  if
-    Body == false ->
-      network:stop(),
-      {stop, "Initialize Failed"};
+start(_Type, _Params) ->
```

Comment:
According to the OTP principles, the application's `start` callback should return `{:ok, TopLevelSupervisorPid}` -- read more [here](https://www.erlang.org/doc/man/application.html#Module:start-2).

This ensures that in case the `statsig_server` crashes, it will be restarted according to the supervisor's restart policy.